packages/astroport/src/asset.rs

```diff
@@ -163,6 +163,28 @@ impl AssetInfo {
             AssetInfo::Token { contract_addr } => contract_addr.as_bytes(),
         }
     }
+
+    pub fn check_lowercase(&self) -> StdResult<()> {
+        match self {
+            AssetInfo::Token { contract_addr } => {
+                if contract_addr.as_str() != contract_addr.as_str().to_lowercase() {
```

Comment:
It would be useful to not only check for uppercase, but to validate the contract address to. Maybe let's update the
`addr_validate_to_lower` to something similar and use it here too?
```
pub fn addr_validate_to_lower(api: &dyn Api, addr: &str) -> StdResult<Addr> {
    if &addr.to_lowercase() != addr {
        return Err(StdError::generic_err(format!(
            "Token address {} should be lowercase",
            addr
        )));
    }

    api.addr_validate(addr)
}
```

Reply:
Ok